llvm/lib/Target/AMDGPU/GCNHazardRecognizer.h

```diff
@@ -94,6 +98,9 @@ class GCNHazardRecognizer final : public ScheduleHazardRecognizer {
   int checkReadM0Hazards(MachineInstr *SMovRel);
   int checkNSAtoVMEMHazard(MachineInstr *MI);
   int checkFPAtomicToDenormModeHazard(MachineInstr *MI);
+  // emit V_NOP instructions. \p WaitStatesNeeded is the number of V_NOPs we
+  // need to insert, negative means not needed.
```

Comment:
Seems silly to use `int` here. Zero already means "not needed" so you don't need negative values.

Reply:
I agree, but all other `check*` functions return `int` as well.